filecoin-project/curio
# Zen/update interface

This PR updates to the latest PDP PRs (still awaiting review)
 - Update pdp_verifier interface to include pass through variables 
 - Update challenge window semantics 
   - New columns to the pdp_proof_sets table to track the challenge window
   - Code in handlers to set up these fields and wait on challenge window being open
   - New code gen for an interface expected of the pdp listener that runs eth calls on the listener contract to get static information about challenge window length and dynamic info about the next challenge window opening

A few things I'm really not sure of
- ~Is it better to be adding columns with ALTER TABLE instead of just changing the schema ?~
- Is there a better way to handle the edge case of the first add? In this PR I just make the nextchallengewindow eth call every time 
- Should we include the ProvingPeriod in pdp_proof_sets table?  It's no longer strictly needed but maybe its nice to keep around? 
- Adding the listener Address twice in pdp_proof_sets and the creates table is probably avoidable.  Is there an easy better way? 
- Are we ok with relying on additional assumptions on the listenerAddr?  Are we ok with it from security perspective? 

I haven't tested this out yet but I'm planning to next week.  Since this code / real databases / SQL is all new to me I think its worth getting your opinion on whether I'm doing something really dumb before I bother running this @magik6k.
